README.md
# colored-log
a colorful `console.log` tool


CHANGELOG.md
# 1.0.1

### Release

- **md:** You can clear the console before output.

### Release 

- **release:** log.info

src/test.ts
import log from "./index"

log.gray(`gray`)
log.red(`red`)
log.green(`green`)
log.yellow(`yellow`)
log.blue(`blue`)
log.magenta(`magenta`)
log.cyan(`cyan`)
log.success(`success`)
log.error(`error`)
log.warn(`warn`)
log.info(`info`)
log(`log`)
// log.clear()

src/index.ts
enum ANSI_COLOR {
  GRAY = '\x1b[90m',
  RED = '\x1b[91m',
  GREEN = '\x1b[92m',
  YELLOW = '\x1b[93m',
  BLUE = '\x1b[94m',
  MAGENTA = '\x1b[95m',
  CYAN = '\x1b[96m',
  RESET = '\x1b[0m',

  SUCCESS = '\x1b[42;30m',
  WARN = '\x1b[43;30m',
  ERROR = '\x1b[41;30m',
  INFO = '\x1b[40;37m',
}
interface Log {
  (data: string): void
  clear(): void
  gray(data: string): void
  red(data: string): void
  green(data: string): void
  yellow(data: string): void
  blue(data: string): void
  magenta(data: string): void
  cyan(data: string): void

  success(data: string): void
  error(data: string): void
  warn(data: string): void
  info(data: string): void
}
const log = console.log as Log

log.clear = console.clear
log.gray = function (data: string) {
  log(ANSI_COLOR.GRAY + data + ANSI_COLOR.RESET)
}
log.red = function (data: string) {
  log(ANSI_COLOR.RED + data + ANSI_COLOR.RESET)
}
log.green = function (data: string) {
  log(ANSI_COLOR.GREEN + data + ANSI_COLOR.RESET)
}
log.yellow = function (data: string) {
  log(ANSI_COLOR.YELLOW + data + ANSI_COLOR.RESET)
}
log.blue = function (data: string) {
  log(ANSI_COLOR.BLUE + data + ANSI_COLOR.RESET)
}
log.magenta = function (data: string) {
  log(ANSI_COLOR.MAGENTA + data + ANSI_COLOR.RESET)
}
log.cyan = function (data: string) {
  log(ANSI_COLOR.CYAN + data + ANSI_COLOR.RESET)
}

log.success = function (data: string) {
  log(ANSI_COLOR.SUCCESS + data + ANSI_COLOR.RESET)
}
log.error = function (data: string) {
  log(ANSI_COLOR.ERROR + data + ANSI_COLOR.RESET)
}
log.warn = function (data: string) {
  log(ANSI_COLOR.WARN + data + ANSI_COLOR.RESET)
}
log.info = function (data: string) {
  log(ANSI_COLOR.INFO + data + ANSI_COLOR.RESET)
}
export default log


lib/index.mjs
// src/index.ts
var log = console.log;
log.clear = console.clear;
log.gray = function(data) {
  log("\x1B[90m" /* GRAY */ + data + "\x1B[0m" /* RESET */);
};
log.red = function(data) {
  log("\x1B[91m" /* RED */ + data + "\x1B[0m" /* RESET */);
};
log.green = function(data) {
  log("\x1B[92m" /* GREEN */ + data + "\x1B[0m" /* RESET */);
};
log.yellow = function(data) {
  log("\x1B[93m" /* YELLOW */ + data + "\x1B[0m" /* RESET */);
};
log.blue = function(data) {
  log("\x1B[94m" /* BLUE */ + data + "\x1B[0m" /* RESET */);
};
log.magenta = function(data) {
  log("\x1B[95m" /* MAGENTA */ + data + "\x1B[0m" /* RESET */);
};
log.cyan = function(data) {
  log("\x1B[96m" /* CYAN */ + data + "\x1B[0m" /* RESET */);
};
log.success = function(data) {
  log("\x1B[42;30m" /* SUCCESS */ + data + "\x1B[0m" /* RESET */);
};
log.error = function(data) {
  log("\x1B[41;30m" /* ERROR */ + data + "\x1B[0m" /* RESET */);
};
log.warn = function(data) {
  log("\x1B[43;30m" /* WARN */ + data + "\x1B[0m" /* RESET */);
};
log.info = function(data) {
  log("\x1B[40;37m" /* INFO */ + data + "\x1B[0m" /* RESET */);
};
var src_default = log;
export {
  src_default as default
};


lib/index.global.js
"use strict";
(() => {
  // src/index.ts
  var log = console.log;
  log.clear = console.clear;
  log.gray = function(data) {
    log("\x1B[90m" /* GRAY */ + data + "\x1B[0m" /* RESET */);
  };
  log.red = function(data) {
    log("\x1B[91m" /* RED */ + data + "\x1B[0m" /* RESET */);
  };
  log.green = function(data) {
    log("\x1B[92m" /* GREEN */ + data + "\x1B[0m" /* RESET */);
  };
  log.yellow = function(data) {
    log("\x1B[93m" /* YELLOW */ + data + "\x1B[0m" /* RESET */);
  };
  log.blue = function(data) {
    log("\x1B[94m" /* BLUE */ + data + "\x1B[0m" /* RESET */);
  };
  log.magenta = function(data) {
    log("\x1B[95m" /* MAGENTA */ + data + "\x1B[0m" /* RESET */);
  };
  log.cyan = function(data) {
    log("\x1B[96m" /* CYAN */ + data + "\x1B[0m" /* RESET */);
  };
  log.success = function(data) {
    log("\x1B[42;30m" /* SUCCESS */ + data + "\x1B[0m" /* RESET */);
  };
  log.error = function(data) {
    log("\x1B[41;30m" /* ERROR */ + data + "\x1B[0m" /* RESET */);
  };
  log.warn = function(data) {
    log("\x1B[43;30m" /* WARN */ + data + "\x1B[0m" /* RESET */);
  };
  log.info = function(data) {
    log("\x1B[40;37m" /* INFO */ + data + "\x1B[0m" /* RESET */);
  };
  var src_default = log;
})();


lib/index.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/index.ts
var src_exports = {};
__export(src_exports, {
  default: () => src_default
});
module.exports = __toCommonJS(src_exports);
var log = console.log;
log.clear = console.clear;
log.gray = function(data) {
  log("\x1B[90m" /* GRAY */ + data + "\x1B[0m" /* RESET */);
};
log.red = function(data) {
  log("\x1B[91m" /* RED */ + data + "\x1B[0m" /* RESET */);
};
log.green = function(data) {
  log("\x1B[92m" /* GREEN */ + data + "\x1B[0m" /* RESET */);
};
log.yellow = function(data) {
  log("\x1B[93m" /* YELLOW */ + data + "\x1B[0m" /* RESET */);
};
log.blue = function(data) {
  log("\x1B[94m" /* BLUE */ + data + "\x1B[0m" /* RESET */);
};
log.magenta = function(data) {
  log("\x1B[95m" /* MAGENTA */ + data + "\x1B[0m" /* RESET */);
};
log.cyan = function(data) {
  log("\x1B[96m" /* CYAN */ + data + "\x1B[0m" /* RESET */);
};
log.success = function(data) {
  log("\x1B[42;30m" /* SUCCESS */ + data + "\x1B[0m" /* RESET */);
};
log.error = function(data) {
  log("\x1B[41;30m" /* ERROR */ + data + "\x1B[0m" /* RESET */);
};
log.warn = function(data) {
  log("\x1B[43;30m" /* WARN */ + data + "\x1B[0m" /* RESET */);
};
log.info = function(data) {
  log("\x1B[40;37m" /* INFO */ + data + "\x1B[0m" /* RESET */);
};
var src_default = log;
